File: LeetCode/Python/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
# TODO: Improve
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        listLen = len(nums)
        for i in range(listLen):
            for j in range(i + 1, listLen):
                if (nums[i] + nums[j]) == target:
                    return [i, j]
# @lc code=end

File: LeetCode/CPP/1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */
#include <vector>
#include <string>
#include <unordered_map>
#include <iostream>
using namespace std;

// @lc code=start
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> present_numbers;
        for (int i = 0; i < nums.size(); i++)
        {
            if (present_numbers[target - nums[i]])
            {
                return {i, present_numbers[target - nums[i]] - 1};
            }
            present_numbers[nums[i]] = i + 1;
        }
        return nums;
    }
};
// @lc code=end